test/test_rpc.py

```diff
@@ -15,16 +15,66 @@
 from common_distributed import MultiProcessTestCase
 from common_utils import load_tests, run_tests, TEST_WITH_ASAN
 from os import getenv
+from typing import (
+    Dict,
+    Sequence,
+)
+from collections import namedtuple
 
 BACKEND = getenv('RPC_BACKEND', RpcBackend.PROCESS_GROUP)
 RPC_INIT_URL = getenv('RPC_INIT_URL', '')
 
-# it is used to test python user defined function over rpc
+
+# classes and functions are used to test python user defined class and
+# methods over rpc
+TensorClass = namedtuple("TensorClass", ["tensors"])
+
+def build_complex_tensors():
+    a = torch.ones(3, 3)
+    b = [a, a]
+    c = [b, b]
+    d = [a, b]
+    e = {a : d}
+    return [a, b, c, d, e]
+
+
+class MyClass:
+    def __init__(self, a):
+        self.a = a
+
+    def my_instance_method(self, b):
+        return self.a + b
+
+    @classmethod
+    def my_class_method(cls, d, e):
+        return d + e
+
+    @staticmethod
+    def my_static_method(f : int):
+        return f > 10
+
+
 def my_function(a, b, c):
     return a + b + c
 
 
-# it is used to test python user defined function over rpc
+def my_tensor_function(a : torch.Tensor, b : torch.Tensor) -> torch.Tensor:
+    return a + b
+
+
+def my_complex_tensor_function(
```

Comment:
One thing that you didn't seem to have tested is UDFs that _return_ a complex value containing tensors inside. We should test that as well as this will exercise a different serialization / deserialization codepath.